errors.go

```diff
@@ -70,3 +70,17 @@ func (t TypeError) Error() string {
 func NewTypeError(expected, actual interface{}) error {
 	return TypeError{fmt.Sprintf("Invalid type for arguments. Expected: %T, actual: %T", expected, actual)}
 }
+
+type CanceledError struct {
+	p string
```

Comment:
Not a big deal, I guess, but by the same logic all error types should be string!